Allow host and port to be passed in via msg.host and msg.port

colinl opened the pull request:
In order to be able to use the node in a subflow with different owserver hosts I have made changes to allow the host and/or port to be passed in via msg.host and msg.port.   I think this may be generally useful and should not affect any existing flows.  I have tested that these work if the fields are left empty in the node, and also that the msg properties over-ride those in the node when they are supplied.
Hopefully this will be considered a worthwhile enhancement.

njh commented on owfs.html:
```diff
@@ -61,8 +62,8 @@
         defaults: {
             name: {value:""},
-            host: {value:"localhost",required:true},
-            port: {value:4304,required:true,validate:RED.validators.number()},
```
What was the reason for changing number() ?

colinl replied:
It seems that the numbers validator does not allow an empty field, which I was a bit surprised about, so I replaced it with a regex which does allow that.  In fact the regex is arguably better anyway as number would presumably allow a decimal point.

njh replied:
Ah, ok. That makes sense.

njh commented on owfs.js:
```diff
@@ -27,46 +27,45 @@ module.exports = function(RED) {
+        node.on("input", function(msg) {
+            var host = msg.host  ||  node.host;
+            var port = msg.port  ||  node.port;
+            //node.log("owserver " + host + ":" + port);
```
comment line left in

colinl replied:
Oops, sorry.  The fact that I had left that in popped into my brain at 3 o'clock this morning for some unfathomable reason.

njh commented on owfs.js:
```diff
@@ -27,46 +27,45 @@ module.exports = function(RED) {
-                }
-            });
-        } else {
-            node.error("missing host and port configuration for owfs");
```
What happens if there is no host or port - does the sensible error message get raised?

colinl replied:
Good point.
At design time it is ok as it shows the missing configuration message so it stops the user scanning for devices.
Will check out what happens at run time.

colinl commented:
I have removed the debug code and added a warning if host and/or port are not available.  The warning shows up in the node-red log and in the debug window in the browser.  Had a bit of difficulty coming up with a concise error message, but hope what I have done is ok.
I did wonder whether it should generate a run time error catchable by a catch block for both this and the missing paths error.  Possibly not necessary as these are both probably design time errors rather than run time.
This is the first time I have updated an existing pull request, hopefully I have done it correctly.

njh commented:
Thanks!
